sklearn/tests/test_base.py

```diff
@@ -246,6 +246,13 @@ def set_params(self, **kwargs):
                        estimator__min_samples_leaf=2)
 
 
+def test_set_params_updates_valid_params():
+    # Check that set_params tries to set SVC().C, not
+    # DecisionTreeClassifier().C
+    pipe = GridSearchCV(DecisionTreeClassifier(), {})
+    pipe.set_params(estimator=SVC(), estimator__C=1.0)
```

Comment:
Could you please extend this test to do `estimator__C=42.0` and then `assert gridsearchcv.estimator.C == 42.0`?